moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -274,8 +274,38 @@ void robot_trajectory::RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::Robot
     {
       trajectory.multi_dof_joint_trajectory.points[i].transforms.resize(mdof.size());
       for (std::size_t j = 0; j < mdof.size(); ++j)
+      {
         tf::transformEigenToMsg(waypoints_[i]->getJointTransform(mdof[j]),
                                 trajectory.multi_dof_joint_trajectory.points[i].transforms[j]);
+        if (waypoints_[i]->hasVelocities())
+        {
+          const std::vector<std::string> names = mdof[j]->getVariableNames();
```

Comment:
please add a check for the *type* of mdof joint and only use the code below if its a planar joint. if you are inclined, add support for a floating joint also. Note that you'd have to check for  ``trans_[x|y|z]`` and ``rot_x, rot_y, rot_z, , rot_w`` which is a quaternion - I'm not sure how a velocity vector would work for that. If you don't want to implement support for that joint type, add a comment with a TODO